.github/workflows/deploy-orchard1-to-azure-app-service.yml

```diff
@@ -3,14 +3,22 @@ name: Deploy Orchard 1 to Azure App Service
 concurrency:
   group: AzureDeployApp
 
+permissions:
+  id-token: write
+  contents: read
+
 on:
   workflow_call:
     secrets:
       CHECKOUT_TOKEN:
         description: >
           The GitHub token to authenticate checkout. Pass in a GitHub personal access token if authenticated submodules
           are used.
-      AZURE_APP_SERVICE_DEPLOYMENT_SERVICE_PRINCIPAL:
+      AZURE_APP_SERVICE_DEPLOYMENT_SERVICE_PRINCIPAL_ID:
+        required: true
+      AZURE_APP_SERVICE_DEPLOYMENT_AZURE_TENANT_ID:
+        required: true
+      AZURE_APP_SERVICE_DEPLOYMENT_AZURE_SUBSCRIPTION_ID:
```

Comment:
Add a description for these. Also in the other workflows.

Reply:
I don't think they each need a description, their names are quite self-explanatory, but I added some context and link to explain how they will be used.

Reply:
Keep in mind that these workflows can be used potentially by anybody, including people with little experience in Azure (including our own colleagues or clients) who need more hand-holding. Since these secrets are part of the public interface of the workflow, they need to be the easiest-to-understand parts of the workflow. Naturally, we're not looking to include a tutorial, but explanations on the level of all the other parameters are necessary, just how we document the published APIs of our libraries.

Reply:
I always keep that in mind and as you can see there's a clear pointer to our login action, so you can see how it's used and the link to the azure/login repo, whose readme has a very detailed explanation of the authentication process. Previously (with the JSON secret) there was no explanation whatsoever.

Reply:
One can surely dig up all the information but having it spelled out is better. You can of course link to the `azure/login` action's docs for further reference too; I'm not saying all the information for these secrets should be right here in the code but some pointers should. (Our `login-to-azure` action is for internal use BTW and we can't expect consumers of this workflow to follow up to that, and then from there to `azure/login`.)

The previous secret not having docs was a mistake that we can now correct.

Reply:
I'm adding code suggestions that I hope will make it clearer what exactly I find useful in this context.

Reply:
I guess it's a matter of where you want to draw line in terms of beginner-friendliness, but OK then.